src/test/java/org/assertj/core/api/map/MapAssert_extractingByKeys_Test.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api.map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.presentation.UnicodeRepresentation.UNICODE_REPRESENTATION;
+import static org.assertj.core.test.AlwaysEqualComparator.ALWAY_EQUALS;
+import static org.assertj.core.test.AlwaysEqualComparator.ALWAY_EQUALS_STRING;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.AbstractListAssert;
+import org.assertj.core.api.MapAssert;
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.util.introspection.PropertyOrFieldSupport;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link MapAssert#extractingByKeys(Object[])}</code>.
+ *
+ * @author Stefano Cordio
+ */
+@DisplayName("MapAssert extractingByKeys(K...)")
+class MapAssert_extractingByKeys_Test {
+
+  private static final Object NAME = "name";
+  private Map<Object, Object> map;
+
+  @BeforeEach
+  void setup() {
+    map = new HashMap<>();
+    map.put(NAME, "kawhi");
+    map.put("age", 25);
+  }
+
+  @Test
+  void should_fail_if_actual_is_null() {
+    // GIVEN
+    Map<Object, Object> map = null;
+    // WHEN
+    AssertionError error = expectAssertionError(() -> assertThat(map).extractingByKeys(NAME, "age"));
+    // THEN
+    then(error).hasMessage(actualIsNull());
+  }
+
+  @Test
+  void should_allow_list_assertions_on_values_extracted_from_given_map_keys() {
+    // WHEN
+    AbstractListAssert<?, List<?>, Object, ObjectAssert<Object>> result = assertThat(map).extractingByKeys(NAME, "age");
+    // THEN
+    result.containsExactly("kawhi", 25);
+  }
```

Comment:
add a tests showing the behavior when providing no keys = will result in an empty list.